### PR TITLE
my intermediate javascript exercise solutions

### PR DESCRIPTION
**Pull request by juliahazer**
fixed mergeObjects functions

### CODE REVIEW

**Comment by juliahazer** [comment]
I added the non-bonus portions of the recursion exercises. I will work on bonus next.

File: testing_exercise/testingSpec.js [hunk 1]
```diff
@@ -1,3 +1,108 @@
...
+	};
+
+	describe('mergeObjects', function(){
+		it('returns an object', function(){
+			expect(mergeObjects(obj1,obj2)).to.be.an('object');
+		});
+
+		it('merges the objects & if same keys, 2nd obj key overrides 1st', function(){
+			expect(mergeObjects(obj1,obj2)).to.deep.equal({
+					name: "Foo",
```
Comment (mmmaaatttttt) [inline testing_exercise/testingSpec.js]: a small thing, but your keys here are over-indented.

File: testing_exercise/testing.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+Write a function called acceptNumbersOnly which takes in any 
+number of arguments and returns true if all of them are numbers. 
+Watch out for NaN - it is a typeof "number"!
+*/
+function acceptNumbersOnly(){
+	for (var i = 0; i < arguments.length; i++){
+		if (typeof(arguments[i]) !== 'number'){
+			return false;
+		}
+		else if(isNaN(arguments[i])){
```
Comment (mmmaaatttttt) [inline testing_exercise/testing.js]: Can you think of a way to refactor this to not use an `else if`?

File: call_apply_bind_exercise/callApplyBind.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+		array-like objects
+	
+*/
+
+/*Write a function called sumEvenArguments which takes all of the arguments 
+passed to a function and returns the sum of the even ones.*/
+
+function sumEvenArguments(...args){
+	return args.reduce(function(acc, el){
+		if (el % 2 === 0) acc += el
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js]: Although this works, I usually put {'s for clarity.  Also, don't forget the ; 

```
if (el % 2 === 0) { acc += el; }
```

File: call_apply_bind_exercise/callApplyBind.js [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+function invokeMax(fn, max){
+  var counter = 0;
+  return function(){
+	  var innerArgs = [].slice.call(arguments);
+	  if (counter < max) {
+		counter++
+		return fn.apply(this, innerArgs)
+	  }
+	  else {
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js]: I like to put the else on the same line:

```
if (blah) {

} else {

}
```

File: call_apply_bind_exercise/callApplyBind.js [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+/*Write a function called invokeMax which accepts a function and a 
+maximum amount. invokeMax should return a function that when called 
+increments a counter. If the counter is greater than the maximum amount, 
+the inner function should return "Maxed Out"*/
+
+function invokeMax(fn, max){
+  var counter = 0;
+  return function(){
+	  var innerArgs = [].slice.call(arguments);
+	  if (counter < max) {
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js]: It looks like you've got a mix of tabs and spaces here. It looks ok on github but ends up being weird on my editor.  I would try changing my settings to always do 2 spaces for a tab character (you can set that up with sublime).

File: call_apply_bind_exercise/callApplyBind.js [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+		else if (guesses > amount){
+			return "You are all done playing!";
+		}
+		else {
+			if (guess === answer){
+				guesses = amount + 1;
+				return "You got it!"
+			}
+			else if (guess > answer){
+				guesses++
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js]: semicolon

File: call_apply_bind_exercise/callApplyBind.js [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+				return "You're too high!"
+			}
+			else {
+				guesses++
```
Comment (tigarcia) [inline call_apply_bind_exercise/callApplyBind.js]: semicolon

File: call_apply_bind_exercise/callApplyBind.js (hunk 3 above)
Comment (juliahazer) [inline call_apply_bind_exercise/callApplyBind.js]: Thanks for this feedback! I will try to use these styling suggestions moving forward.

**Pull request closed without merging**